packages/api/src/controllers/access-control.ts

```diff
@@ -40,83 +44,181 @@ app.post(
 
     if (user.suspended || ("suspended" in content && content.suspended)) {
       const contentLog = JSON.stringify(JSON.stringify(content));
-      const userLog = JSON.stringify(JSON.stringify(user));
-
       console.log(`
-        access-control: gate: disallowing access for contentId=${content.id} playbackId=${playbackId}, user ${user.id} is suspended, content=${contentLog}, user=${userLog}
+        access-control: gate: disallowing access for contentId=${content.id} playbackId=${playbackId}, user=${user.id} is suspended, content=${contentLog}
       `);
       throw new NotFoundError("Content not found");
     }
 
     const playbackPolicyType = content.playbackPolicy?.type ?? "public";
 
-    if (playbackPolicyType === "public") {
-      res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
-      res.status(204);
-      return res.end();
-    } else if (playbackPolicyType === "jwt") {
-      if (!req.body.pub) {
-        console.log(`
-          access-control: gate: no pub provided for playbackId ${playbackId}, disallowing playback
-        `);
-        throw new ForbiddenError("Content is gated and requires a public key");
-      }
-
-      const query = [];
-      query.push(sql`signing_key.data->>'publicKey' = ${req.body.pub}`);
-      const [signingKeyOutput] = await db.signingKey.find(query, {
-        limit: 2,
-      });
-
-      if (signingKeyOutput.length == 0) {
-        console.log(`
-          access-control: gate: content with playbackId ${playbackId} is gated but corresponding public key not found for key ${req.body.pub}, disallowing playback
-        `);
-        throw new ForbiddenError(
-          "Content is gated and corresponding public key not found"
+    switch (playbackPolicyType) {
+      case "public":
+        res.set("Cache-Control", "max-age=120,stale-while-revalidate=600");
```

Comment:
How about we always set this header for the response? Meaning do it in the beginning of the controller. If we reply with a 5xx status code (like network error idk), clients will still avoid caching it anyway as they cache only 2xx and 4xx. ([catalyst only caches 2xx](https://github.com/livepeer/catalyst/blob/main/cmd/catalyst-node/handlers/access-control/access-control.go#L220))

Will avoid having paths where we forget to set it in the code tho wanted it and getting /gate api bursts from invalid playback URLs.